Code/Tools/AssetProcessor/native/resourcecompiler/RCQueueSortModel.cpp

```diff
@@ -55,12 +56,61 @@ namespace AssetProcessor
+                    // Asset A finishes processing job, and outputs the product.
+                    // Intermediate A has not yet been scanned and discovered by Asset Processor, so it's not in flight or in the queue yet.
+                    // Asset Processor goes to pull the next job in the queue. Asset B still technically has a missing job dependency on the intermediate asset output.
+                    // Asset B gets pulled from the queue to process here, even though Intermediate A hasn't run yet, because Intermediate A hasn't gone into the queue yet.
+                    // This happened with FBX files and a dependency on an intermediate asset materialtype, before Common platform jobs were made higher priority than host platform jobs.
+                    // Why not just check if the target file exists at this point? Because the job key has to match up.
+                    // When a check was added here to see if all the dependency files existed, other material jobs started to end up in the queue indefinitely
+                    // because they had a dependency on a file that existed but a job key that did not exist for that file.
+                    continue;
+                }
+
+                if (actualJob->HasMissingSourceDependency())
+                {
+                    AZ_Warning(
+                        AssetProcessor::ConsoleChannel,
+                        false,
+                        "No job was found to match the job dependency criteria declared by file %s.\n"
+                        "This may be due to a mismatched job key.\n"
+                        "Job ordering will not be guaranteed and could result in errors or unexpected output.",
+                        actualJob->GetJobEntry().GetAbsoluteSourcePath().toUtf8().constData());
+
+                    // Print out all dependencies, to help track down the dependency gap.
+                    AZ_Printf(
+                        AssetProcessor::ConsoleChannel,
+                        "All dependencies for file %s\n",
+                        actualJob->GetJobEntry().GetAbsoluteSourcePath().toUtf8().constData());
+                    for (const JobDependencyInternal& jobDependencyInternal : actualJob->GetJobDependencies())
+                    {
+                        // Include the source name so if only this log line is seen, it's still useful.
+                        AZ_Printf(AssetProcessor::ConsoleChannel, "\tSource %s has dependency - File: %s, JobKey: %s, Platform: %s",
+                            actualJob->GetJobEntry().GetAbsoluteSourcePath().toUtf8().constData(),
+                            jobDependencyInternal.m_jobDependency.m_sourceFile.m_sourceFileDependencyPath.c_str(),
+                            jobDependencyInternal.m_jobDependency.m_jobKey.c_str(),
+                            jobDependencyInternal.m_jobDependency.m_platformIdentifier.c_str());
+                    }
```

Comment:
Is this intended to be committed?  If so, should this be using AZ_Trace?

Reply:
Yeah, that's intentional. It's additional context for the wanring a few lines up. I'll switch it to Trace. I thought we weren't using TracePrintf anymore and used Printf now, though? Or do I have that backward?

Reply:
I removed this printf, it was causing an unused variable issue on Jenkins, and the printf isn't important enough to be worth wrapping it with unused or #ifdef.